backend/danswer/server/query_and_chat/chat_backend.py

```diff
@@ -272,20 +273,12 @@ def delete_chat_session_by_id(
 
 
 @router.post("/send-message")
-def handle_new_chat_message(
+async def handle_new_chat_message(
```

Comment:
is it possible to do this in a non-async way? I think this could have some serious performance consequences (e.g. since none of the DB / OpenAI calls are async, this could block the main process preventing other requests from going through)

Reply:
Might be slightly tricky considering the asynchronous nature of the callback/`StreamingResponse` - happy middle ground might be to see the performance impact and then consider doing something like move blocking operations to  a separate thread?

Reply:
Specifically, request.is_disconnected, from the FastAPI docs, returns a coroutine